### PR TITLE
Will provide support for ephemeral block_device_mappings

### PR DESCRIPTION
**Pull request by erkolson**
Resolves issue #172 

Example of ephemeral drives used on the same instance in conjunction with an ebs device, in kitchen.yml:
```yaml
      block_device_mappings:
        - ebs_device_name: /dev/xvda
          ebs_volume_type: gp2
          ebs_volume_size: 8
          ebs_delete_on_termination: true
        - device_name: /dev/xvdb
          virtual_name: ephemeral0
        - device_name: /dev/xvdc
          virtual_name: ephemeral1
```

### CODE REVIEW

**Comment by kingpong** [comment]
+1

**Comment by grosendorf** [comment]
:+1:

**Comment by dougalb** [comment]
+1

**Comment by dzabel** [comment]
Would be great to see this in next release

**Comment by cheeseplus** [comment]
Could we get this rebased so the PR gets tested using the latest travis configs?

**Comment by erkolson** [comment]
@cheeseplus, I think that covers it.

File: lib/kitchen/driver/ec2.rb [hunk 1]
```diff
@@ -116,11 +116,15 @@ def self.validation_warn(driver, old_key, new_key)
       validations[:block_device_mappings] = lambda do |_attr, val, _driver|
         unless val.nil?
           val.each do |bdm|
-            unless bdm.keys.include?(:ebs_volume_size) &&
+            if bdm.keys.include?(:ebs_volume_size) &&
```
Comment (jkeiser) [inline lib/kitchen/driver/ec2.rb]: Why change  unless to if here? The "empty if" pattern is a little hard to read.

**Comment by jkeiser** [comment]
@erkolson I have submitted #223 which fixes your problem by allowing arbitrary values to be passed to `block_device_mappings`. To do what you suggested in your example above, you can do this:

```yaml
      block_device_mappings:
        - device_name: /dev/xvda
          ebs:
            volume_type: gp2
            volume_size: 8
            delete_on_termination: true
        - device_name: /dev/xvdb
          virtual_name: ephemeral0
        - device_name: /dev/xvdc
          virtual_name: ephemeral1
```

**Comment by jkeiser** [comment]
If you get some time to try that out, I'd super appreciate it! If it works for you, we can close this one.

**Comment by jkeiser** [comment]
I believe this is fixed in master now (with the aforementioned patch having been merged).

**Comment by erkolson** [comment]
I don't see it in master... how should I test?

**Comment by jkeiser** [comment]
It was this one: https://github.com/test-kitchen/kitchen-ec2/commit/5e92115b69b26beaaf33a22be56d43ded7dfde81

You can either `rake install` from latest master, or you can `gem install kitchen-ec2-1.0.0.beta.1`.

**Comment by erkolson** [comment]
Thanks, I'll test on Monday.  I've been working on a production issue all night and don't feel like doing much more today :-/

**Comment by TheLinuxNinja** [comment]
I just came across this while trying to find why some of our .kitchen.yml files don't work anymore. Seems that we were able to have ephemerals at one time, then they were broken. I see the last activity here is over a month ago. Where does it stand?

**Comment by erkolson** [comment]
@jkeiser, Sorry for the delay.  I was able to test 1.0.0.beta.1 yesterday.  It works for me.  Is this already in the 1.0.0 release?

**Comment by jkeiser** [comment]
1.0.0 should have this, yep!

**Comment by roshi619** [comment]
@jkeiser .. it seems like this is not yet available within 1.0.0. Could you check and merge accordingly.

**Pull request closed without merging**